File: administrador_plants/src/app/models/Mapa.model.ts
export interface Mapa {
  id: number
  id_planta: number
  zona: string
  latitud: number
  longitud: number
  url_imagen: string
  estatus: number
  created_at: string
  info_plantas: InfoPlantas
}

export interface InformacionPlanta {
  id: number
  nombre_planta: string
  nombre_cientifico: string
  toxicidad: any
  año: number
  familia: string
  nombres_comunes: NombresComunes
  distribucion: Distribucion
  colores: any
  humedad_atmosferica?: string
  cantidad_luz: string
  meses_crecimiento: any
  genero: string
  estatus: number
  id_planta: number
  zona: string
  latitud: number
  longitud: number
  url_imagen: string
  created_at: string
}

export interface InfoPlantas {
  id: number
  nombre_planta: string
  nombre_cientifico: string
  toxicidad: any
  año: number
  familia: string
  nombres_comunes: NombresComunes
  distribucion: Distribucion
  colores: any
  humedad_atmosferica?: string
  cantidad_luz: string
  meses_crecimiento: any
  genero: string
  estatus: string
}

export interface NombresComunes {
  dan?: string[]
  eng: string[]
  tur?: string[]
  deu: string[]
  spa: string[]
  ita?: string[]
  heb?: string[]
  fra: string[]
  swa: string[]
  por: string[]
  ell?: string[]
  hun?: string[]
  ces?: string[]
  swe: string[]
  nld?: string[]
  sme?: string[]
  cym?: string[]
  en: string[]
  af: string[]
  sq: string[]
  am: string[]
  ar: string[]
  hy: string[]
  az: string[]
  eu: string[]
  be: string[]
  bn: string[]
  bs?: string[]
  br: string[]
  bg: string[]
  my: string[]
  ca: string[]
  zh: string[]
  cv: string[]
  kw?: string[]
  hr: string[]
  cs: string[]
  da: string[]
  nl: string[]
  eo: string[]
  et: string[]
  fi: string[]
  fr: string[]
  gl: string[]
  ka: string[]
  de: string[]
  el: string[]
  ha: string[]
  he: string[]
  hi: string[]
  hu: string[]
  io: string[]
  id: string[]
  ie?: string[]
  ga: string[]
  it: string[]
  ja?: string[]
  jv: string[]
  ko: string[]
  ku: string[]
  lv: string[]
  lt: string[]
  lb: string[]
  mk: string[]
  mg: string[]
  ms: string[]
  ml: string[]
  mr: string[]
  mn: string[]
  nv: string[]
  ne: string[]
  se?: string[]
  no: string[]
  oc?: string[]
  os?: string[]
  pa: string[]
  ps: string[]
  fa: string[]
  pl: string[]
  pt: string[]
  "pt-br": string[]
  qu: string[]
  ro: string[]
  ru: string[]
  sc?: string[]
  gd?: string[]
  sr: string[]
  sn?: string[]
  sd: string[]
  sk: string[]
  es: string[]
  su: string[]
  sw: string[]
  sv: string[]
  tl: string[]
  "zh-tw": string[]
  ta: string[]
  tt: string[]
  te: string[]
  th: string[]
  bo: string[]
  tr: string[]
  ug?: string[]
  uk: string[]
  ur: string[]
  vi: string[]
  cy: string[]
  msa?: string[]
  ak?: string[]
  bi?: string[]
  dv?: string[]
  fj?: string[]
  gn?: string[]
  gu?: string[]
  ht?: string[]
  is?: string[]
  ig?: string[]
  ia?: string[]
  kn?: string[]
  kk?: string[]
  km?: string[]
  lo?: string[]
  li?: string[]
  ln?: string[]
  mt?: string[]
  gv?: string[]
  mi?: string[]
  or?: string[]
  sa?: string[]
  si?: string[]
  so?: string[]
  ty?: string[]
  to?: string[]
  "zh-hant"?: string[]
  fy?: string[]
  wo?: string[]
  yi?: string[]
  yo?: string[]
}

export interface Distribucion {
  native: Native[]
  introduced: Introduced[]
}

export interface Native {
  id: number
  name: string
  slug: string
  tdwg_code: string
  tdwg_level: number
  species_count: number
  links: Links
}

export interface Links {
  self: string
  plants: string
  species: string
}

export interface Introduced {
  id: number
  name: string
  slug: string
  tdwg_code: string
  tdwg_level: number
  species_count: number
  links: Links2
}

export interface Links2 {
  self: string
  plants: string
  species: string
}

File: plants_movil/lib/models/Mapa.model.dart
class Mapa {
  int? id;
  int? idPlanta;
  String? zona;
  double? latitud;
  double? longitud;
  String? urlImagen;
  int? estatus;
  String? createdAt;
  InfoPlantas? infoPlantas;

  Mapa(
      {this.id,
      this.idPlanta,
      this.zona,
      this.latitud,
      this.longitud,
      this.urlImagen,
      this.estatus,
      this.createdAt,
      this.infoPlantas});

  Mapa.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    idPlanta = json['id_planta'];
    zona = json['zona'];
    latitud = json['latitud'];
    longitud = json['longitud'];
    urlImagen = json['url_imagen'];
    estatus = json['estatus'];
    createdAt = json['created_at'];
    infoPlantas = json['info_plantas'] != null
        ? InfoPlantas.fromJson(json['info_plantas'])
        : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['id_planta'] = idPlanta;
    data['zona'] = zona;
    data['latitud'] = latitud;
    data['longitud'] = longitud;
    data['url_imagen'] = urlImagen;
    data['estatus'] = estatus;
    data['created_at'] = createdAt;
    if (infoPlantas != null) {
      data['info_plantas'] = infoPlantas!.toJson();
    }
    return data;
  }
}

class InfoPlantas {
  int? id;
  String? nombrePlanta;
  String? nombreCientifico;
  String? urlImagen;
  String? toxicidad;
  bool? comestible;
  bool? vegetable;
  int? aO;
  String? familia;
  NombresComunes? nombresComunes;

  List<String>? colores;
  String? humedadAtmosferica;
  String? cantidadLuz;
  List<String>? mesesCrecimiento;
  String? genero;
  String? estatus;
  String? createdAt;

  InfoPlantas(
      {this.id,
      this.nombrePlanta,
      this.nombreCientifico,
      this.urlImagen,
      this.toxicidad,
      this.comestible,
      this.vegetable,
      this.aO,
      this.familia,
      this.colores,
      this.humedadAtmosferica,
      this.cantidadLuz,
      this.mesesCrecimiento,
      this.nombresComunes,
      this.genero,
      this.estatus,
      this.createdAt});

  InfoPlantas.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nombrePlanta = json['nombre_planta'];
    nombreCientifico = json['nombre_cientifico'];
    urlImagen = json['url_imagen'];
    toxicidad = json['toxicidad'];
    comestible = json['comestible'];
    vegetable = json['vegetable'];
    aO = json['año'];
    if (json["colores"] != null) {
      colores = json['colores'].cast<String>();
    }
    if (json["meses_crecimiento"] != null) {
      mesesCrecimiento = json['meses_crecimiento'].cast<String>();
    }
    if (json["nombres_comunes"] != null) {
      nombresComunes = NombresComunes.fromJson(json["nombres_comunes"]);
    }

    familia = json['familia'];
    humedadAtmosferica = json['humedad_atmosferica'];
    cantidadLuz = json['cantidad_luz'];
    genero = json['genero'];
    estatus = json['estatus'];
    createdAt = json['created_at'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['nombre_planta'] = nombrePlanta;
    data['nombre_cientifico'] = nombreCientifico;
    data['url_imagen'] = urlImagen;
    data['toxicidad'] = toxicidad;
    data['comestible'] = comestible;
    data['vegetable'] = vegetable;
    data['año'] = aO;
    data['familia'] = familia;
    data['colores'] = colores;
    data['humedad_atmosferica'] = humedadAtmosferica;
    data['cantidad_luz'] = cantidadLuz;
    data['meses_crecimiento'] = mesesCrecimiento;
    data['genero'] = genero;
    data['estatus'] = estatus;
    data['created_at'] = createdAt;
    data['nombres_comunes'] = nombresComunes;
    return data;
  }
}

class NombresComunes {
  List<String>? spa;

  NombresComunes({this.spa});

  NombresComunes.fromJson(Map<String, dynamic> json) {
    spa = (json['spa'] != null) ? json['spa'].cast<String>() : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};

    data['spa'] = spa;
    return data;
  }
}

class Distribucion {
  List<Native>? native;

  Distribucion({this.native});

  Distribucion.fromJson(Map<String, dynamic> json) {
    if (json['native'] != null) {
      native = <Native>[];
      json['native'].forEach((v) {
        native!.add(Native.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    if (native != null) {
      data['native'] = native!.map((v) => v.toJson()).toList();
    }

    return data;
  }
}

class Native {
  int? id;
  String? name;
  String? slug;
  String? tdwgCode;
  int? tdwgLevel;
  int? speciesCount;
  Links? links;

  Native(
      {this.id,
      this.name,
      this.slug,
      this.tdwgCode,
      this.tdwgLevel,
      this.speciesCount,
      this.links});

  Native.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    slug = json['slug'];
    tdwgCode = json['tdwg_code'];
    tdwgLevel = json['tdwg_level'];
    speciesCount = json['species_count'];
    links = json['links'] != null ? Links.fromJson(json['links']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['name'] = name;
    data['slug'] = slug;
    data['tdwg_code'] = tdwgCode;
    data['tdwg_level'] = tdwgLevel;
    data['species_count'] = speciesCount;
    if (links != null) {
      data['links'] = links!.toJson();
    }
    return data;
  }
}

class Links {
  String? self;
  String? plants;
  String? species;

  Links({this.self, this.plants, this.species});

  Links.fromJson(Map<String, dynamic> json) {
    self = json['self'];
    plants = json['plants'];
    species = json['species'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['self'] = self;
    data['plants'] = plants;
    data['species'] = species;
    return data;
  }
}
